packages/quantic/force-app/main/default/lwc/quanticResultsPerPage/quanticResultsPerPage.js

```diff
@@ -51,7 +51,13 @@ export default class QuanticResultsPerPage extends LightningElement {
     return this._initialChoice;
   }
   set initialChoice(value) {
-    this._initialChoice = value;
+    if (Number(value) >= 0) {
+      this._initialChoice = value;
+    } else {
+      console.warn(
```

Comment:
Also in the consistency topic,
I find it a little bit strange that when initialChoice is <= 0 we log a warning and we default to the first value in the choices displayed without breaking the component.
But when one of the values of `choicesDisplayed ` is negative number, we log an error and we break the component by displaying the error message.

For me it seems that sometimes we are being tolerant and sometimes we are not, I would be be more comfortable if we handled every property validation the same way,  this way our library will always follow one pattern when it comes to property validation providing a better developer experience, what do you think?

Reply:
Well we do have a fallback if you set a wrong `initialChoice` but we possibly don't have a fallback if you set a negative or non-number in the `choicesToDisplay`.

That was my logic anyway. It seemed to make sense to "recover" from a wrong `initialChoice` but not to ignore potentially 1 or more values from the `choicesToDisplay`.

I do agree that we can also be more strict, if you pass a wrong value anywhere, we can simply error out instead.
I'll change it for consistency indeed, after your message, I think that we don't really gain anything from "recovering" from a bad value in the `initialChoice`.

Reply:
@mmitiche let me know if it makes more sense now.

Updated the description